spec/integrations/post_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+RSpec.describe 'posts#index', type: :feature do
+  describe 'Post' do
+    before(:each) do
+      @user1 = User.create(name: 'John', photo: 'https://picsum.photos/seed/picsum/536/354', bio: 'bio')
+      @user1.save!
+      visit root_path
+      @post1 = Post.create(title: 'First Post', text: 'This is my first post', likes_counter: 3, author: @user1)
+      @post2 = Post.create(title: 'Second Post', text: 'This is my second post', author: @user1)
+      @post3 = Post.create(title: 'Third Post', text: 'This is my third post', author: @user1)
+      @comment1 = Comment.create(text: 'Comment first', author: User.first, post: @post1)
+      @comment2 = Comment.create(text: 'Comment another', author: User.first, post: @post1)
+      @comment3 = Comment.create(text: 'Comment new', author: User.first, post: @post1)
+    end
+
+    it "shows user's profile picture" do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_css('img[src*="https://picsum.photos/seed/picsum/536/354"]')
+    end
+
+    it 'shows the users username' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('John')
+    end
+
+    it 'shows number of posts of user has written' do
+      visit(user_posts_path(@user1.id))
+      expect(@user1.posts_counter).to eql(3)
+    end
+
+    it 'shows number of posts by user' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('posts: 3')
+    end
+
+    it 'shows posts title' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('Index Posts')
+    end
+
+    it 'can see some of the post detail' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('This is my first post')
+    end
+
+    it 'can see the first comment on a post' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('Comment first')
+    end
+
+    it 'can see how many comments a post has.' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('Comments: 3')
+    end
+
+    it 'can see how many likes a post has.' do
+      visit(user_posts_path(@user1.id))
+      expect(page).to have_content('Likes: 3')
+    end
+  end
+end
```

Comment:
**For Post index**

- [ ] I can see that you might have forgotten to add specs for "I can see a section for pagination if there are more posts than fit on the view."

![image](https://user-images.githubusercontent.com/86318284/205822951-0d8ef2a3-2ac9-4d98-b140-e9319a625360.png)


- [ ] - [ ] I can see that you might have forgotten to add specs for "When I click on a post, it redirects me to that post's show page"

![image](https://user-images.githubusercontent.com/86318284/205823000-3082f2c9-c114-4c29-9040-6307243c23c0.png)

**Kindly add them**

Reply:
Thank you very much for your observations. I have finished applying them.